code/game/objects/structures/ghost_role_spawners.dm

```diff
@@ -595,3 +595,45 @@
 
 /obj/effect/mob_spawn/human/pirate/gunner
 	rank = "Gunner"
+
+/obj/effect/mob_spawn/human/ghostcafe
+	name = "ghost cafe sleeper"
+	uses = -1
+	icon = 'icons/obj/machines/sleeper.dmi'
+	icon_state = "sleeper"
+	mob_name = "a ghost cafe visitor"
+	roundstart = FALSE
+	anchored = TRUE
+	density = FALSE
+	death = FALSE
+	assignedrole = "Ghost Cafe Visitor"
+
+/obj/effect/mob_spawn/human/ghostcafe/special(mob/living/carbon/human/new_spawn)
+	if(new_spawn.client)
+		new_spawn.fully_replace_character_name(newname = new_spawn.client.prefs.real_name)
+		new_spawn.nameless = new_spawn.client.prefs.nameless
+		new_spawn.gender = new_spawn.client.prefs.gender
+		new_spawn.dna.features = new_spawn.client.prefs.features.Copy()
+		new_spawn.flavor_text = new_spawn.dna.features["flavor_text"]
+		new_spawn.dna.custom_species = new_spawn.client.prefs.custom_species
+		new_spawn.eye_color = new_spawn.client.prefs.eye_color
+		new_spawn.hair_color = new_spawn.client.prefs.facial_hair_color
+		new_spawn.skin_tone = new_spawn.client.prefs.skin_tone
+		new_spawn.facial_hair_style = new_spawn.client.prefs.facial_hair_style
+		new_spawn.hair_color = new_spawn.client.prefs.hair_color
+		new_spawn.hair_style = new_spawn.client.prefs.hair_style
+		new_spawn.saved_socks = new_spawn.client.prefs.socks
+		new_spawn.socks = new_spawn.client.prefs.socks
+		new_spawn.socks_color = new_spawn.client.prefs.socks_color
+		new_spawn.saved_undershirt = new_spawn.client.prefs.undershirt
+		new_spawn.undershirt = new_spawn.client.prefs.undershirt
+		new_spawn.shirt_color = new_spawn.client.prefs.shirt_color
+		new_spawn.saved_underwear = new_spawn.client.prefs.underwear
+		new_spawn.underwear = new_spawn.client.prefs.underwear
+		new_spawn.undie_color = new_spawn.client.prefs.undie_color
+		new_spawn.set_species(new_spawn.client.prefs.pref_species, icon_update=1)
+		new_spawn.give_genitals(TRUE)
```

Comment:
Any reason for doing all of this when
https://github.com/Citadel-Station-13/Citadel-Station-13/blob/765832373d490468e349d96ec5e2dbdfbc593e9d/code/modules/client/preferences.dm#L2252
could be used instead? (there is also a separate modular_citadel bit for the extra bits)

Reply:
That's a proc I wasn't even aware of.
I've looked at the virtual reality code (which copies shit from pre-existing mob) and how turning a ghost into human worked in the admin panel, and neither of those fit my purposes.
I'll test it now.